mycroft/skills/settings.py

```diff
@@ -12,571 +12,383 @@
+    from mycroft.skill.settings import SkillSettings
 
         s = SkillSettings('./settings.json', 'ImportantSettings')
-        s['meaning of life'] = 42
-        s['flower pot sayings'] = 'Not again...'
-        s.store()  # This happens automagically in a MycroftSkill
+        s.skill_settings['meaning of life'] = 42
+        s.skill_settings['flower pot sayings'] = 'Not again...'
+        s.save_settings()  # This happens automagically in a MycroftSkill
 """
-
 import json
-import hashlib
 import os
 import re
-import time
-from os.path import isfile, join
-from requests.exceptions import RequestException, HTTPError
-from threading import Timer, Thread
-
-from msm import SkillEntry
+from pathlib import Path
+from threading import Timer
 
 from mycroft.api import DeviceApi, is_paired
-from mycroft.util.log import LOG
+from mycroft.configuration import Configuration
+from mycroft.messagebus.message import Message
 from mycroft.util import camel_case_split
-from mycroft.configuration import ConfigurationManager
+from mycroft.util.log import LOG
 from .msm_wrapper import build_msm_config, create_msm
 
+ONE_MINUTE = 60
 
-msm = None
-msm_creation_time = 0
 
+def get_local_settings(skill_dir, skill_name) -> dict:
+    """Build a dictionary using the JSON string stored in settings.json."""
+    settings_path = Path(skill_dir).joinpath('settings.json')
+    if settings_path.is_file():
+        with open(str(settings_path)) as settings_file:
+            try:
+                skill_settings = json.load(settings_file)
+            # TODO change to check for JSONDecodeError in 19.08
+            except Exception:
+                log_msg = 'Failed to load {} settings from settings.json'
+                LOG.exception(log_msg.format(skill_name))
+                skill_settings = {}
+    else:
+        skill_settings = {}
 
-def build_global_id(directory, config):
-    """ Create global id for the skill.
-
-    TODO: Handle dirty skill
-
-    Arguments:
-        directory:  skill directory
-        config:     config for the device to fetch msm setup
-     """
-    # Update the msm object if it's more than an hour old
-    global msm
-    global msm_creation_time
-    if msm is None or time.time() - msm_creation_time > 60 * 60:
-        msm_creation_time = time.time()
-        LOG.info('instantiating msm...')
-        msm_config = build_msm_config(config)
-        msm = create_msm(msm_config)
-        LOG.info('msm instantiation complete')
-
-    skill = SkillEntry.from_folder(directory, msm)
-    # If modified prepend the device uuid
-    LOG.info('building skill gid for ' + skill.name)
-    return skill.skill_gid, skill.meta_info.get('display_name')
+    return skill_settings
 
 
-def display_name(name):
-    """ Splits camelcase and removes leading/trailing Skill. """
-    name = re.sub(r'(^[Ss]kill|[Ss]kill$)', '', name)
-    return camel_case_split(name)
+def save_settings(skill_dir, skill_settings):
+    """Save skill settings to file."""
```

Comment:
Needs to handle the Settings class specially...

Something along the lines of
```python
def save_settings(skill_dir, skill_settings):                                   
    """Save skill settings to file."""                                          
    settings_path = Path(skill_dir).joinpath('settings.json')                   
                                                                                
    if isinstance(skill_settings, Settings):                                    
        settings = skill_settings.to_dict()                                     
    elif isinstance(skill_settings, dict):                                      
        settings = skill_settings                                               
    else:                                                                       
        raise ValueError('skill_settings should be a dict, not '                
                         '{}'.format(type(skill_settings)))                     
    with open(str(settings_path), 'w') as settings_file:                        
        json.dump(settings, settings_file)
```

Reply:
Why is this necessary?  When I call this method from the settings class, I pass the _settings attribute, which is a dictionary.

Reply:
In the future we intend to just pass the MycroftSkill.settings member so we should start doing that directly. Which will show people how to use the function as intended.

Also accessing a _attribute from outside the class should really not be done